server/api/image.py

```diff
@@ -147,6 +151,28 @@ def download(self, image, params):
         return file_stream
 
     @describeRoute(
+        Description('Return histograms of image metadata.')
+        .param('filter',
+               'Get the histogram after the results of this filter. ' +
+               'TODO: describe our filter grammar (an AST tree).',
+               required=False)
+        .param('limit', 'Result set size limit. Setting to 0 will create ' +
+               'a histogram using all the matching items (default=0).',
+               required=False, dataType='int')
+        .param('offset', 'Offset into result set (default=0).',
+               required=False, dataType='int')
+        .errorResponse()
+    )
+    @access.public
+    def histogram(self, params):
+        # TODO: ideally we should access images via the image route, but it
```

Comment:
Because `Image` is a subtype of `Item`, you can get all Images with:
```
imagesCollection = self.model('collection').collection.find_one({'name': 'Lesion Images'})
allImagesCursor = self.model('item').collection.find({'baseParentId': imagesCollection['_id']})
```
Of course, just calling `self.model('image', 'isic_archive').find()` does all this for you.

Is there any way to pre-filter things here, and pass the Images cursor into `histogramUtility.getHistograms`, instead of passing an unfiltered collection?